sklearn/decomposition/tests/test_sparse_pca.py

```diff
@@ -315,6 +315,32 @@ def test_spca_early_stopping(global_random_seed):
     assert model_early_stopped.n_iter_ < model_not_early_stopped.n_iter_
 
 
+def test_equivalence_components_pca_spca():
+    """Check the equivalence of the components found by PCA and SparsePCA.
+
+    Non-regression test for:
+    https://github.com/scikit-learn/scikit-learn/issues/23932
+    """
+    rng = np.random.RandomState(0)
```

Comment:
I think this test is seed dependent. For me this test passes locally on `main` with `seed=0` and fails with `seed=1`.

Reply:
It is weird, I cannot reproduce on my machine. Let me try with a global random seed.

Reply:
In `main`, it should fail because `SparsePCA` components are not deterministic. The SVD flip makes sure that the components are deterministic.

In any case, I just add the global random seed such that we make sure that it is always working, independently of the platform.